src/hashstore/hashstore.py

```diff
@@ -10,44 +10,133 @@
 from contextlib import closing
 from tempfile import NamedTemporaryFile
 from collections import namedtuple
+from hashstore.hashstore_interface import HashStoreInterface
+from hashstore.hashstore_config import (
+    DIR_DEPTH,
+    DIR_WIDTH,
+    SYSMETA_NS,
+    DEFAULT_ALGO_LIST,
+    OTHER_ALGO_LIST,
+)
+
+
+class HashStoreFactory:
+    """A factory class for creating `HashStore`-like objects (classes
+    that implement the 'hashstore_interface' abstract methods)
+
+    This factory class provides a method to retrieve a `hashstore` object
+    based on the specified store type (ex. FileHashStore). It supports the
+    creation of different types of hash stores by mapping store types to
+    specific implementations.
+    """
+
+    def __init__(self):
+        """Initialize the HashStoreFactory with default config values"""
+        self.dir_depth = DIR_DEPTH
+        self.dir_width = DIR_WIDTH
+        self.sysmeta_ns = SYSMETA_NS
+        self.default_algo_list = DEFAULT_ALGO_LIST
+        self.other_algo_list = OTHER_ALGO_LIST
+
+    def get_hashstore(self, store_path, hashstore_type):
+        """Get a `HashStore`-like object based on the specified store type.
+
+        Args:
+            root (str): The root directory for the hash store.
+            hashstore_type (str): The type of `HashStore` to retrieve.
+
+        Returns:
+            HashStore: A hash store object based on the given store type.
+
+        Raises:
+            ValueError: If the given store_type is not supported.
+        """
+        hashstore_type.lower()
+        if hashstore_type == "filehashstore":
+            return FileHashStore(
+                root=store_path,
+                depth=self.dir_depth,
+                width=self.dir_width,
```

Comment:
Why do these have to be passed to FileHashStore, instead of FileHashStore grabbing them from the config file itself? 

AFAICT, the values are very specific to the FileHashStore implementation of the HashStoreInterface, so having HashStoreFactory retrieve them from the config and pass them in like this, means you're leaking your implementation details into other places (that shouldn't really "need to know").

Adding a new type of hashstore implementation will therefore necessitate custom code changes in multiple places, instead of just plugging in a new one and adding a line to a config.

Reply:
I have since created (moved) the 'FileHashStore' module and refactored the __init__() method so that it only takes a `root` parameter. `root` is set to `None` and if it is not supplied, the default path will be based on the config file `STORE_PATH` which is currently `"var/filehashstore/`. A larger conversation regarding the config mechanism will take place and this will be revised once we have that discussion with the backend team.